packages/flutter/lib/src/material/dialog.dart

```diff
@@ -909,16 +912,19 @@ class AlertDialog extends StatelessWidget {
       );
     }
 
-    return Dialog(
-      backgroundColor: backgroundColor,
-      elevation: elevation,
-      shadowColor: shadowColor,
-      surfaceTintColor: surfaceTintColor,
-      insetPadding: insetPadding,
-      clipBehavior: clipBehavior,
-      shape: shape,
-      alignment: alignment,
-      child: dialogChild,
+    return Semantics(
+      role: SemanticsRole.alertDialog,
+      child: Dialog(
```

Comment:
To avoid having two roles set on the dialog that need to be resolve by merging or something, can we instead have the `Dialog` constructor take an optional `semanticsRole` argument (similar to `semanticLabel` that many of our widgets accept):

```
Dialog({
  SemanticsRole semanticRole = SemanticsRole.dialog,
  ... other args ...
});
```